cmd/loadTests.go

```diff
@@ -1331,3 +1606,38 @@
 
 	return deploymentCompleted && deploymentReachedDesiredState, lastUpdateTime
 }
+
+// TraceRecord represents a single trace record
+type TraceRecord struct {
+	FunctionName string                 `json:"functionName"`
+	Timestamp    time.Time              `json:"timestamp"`
+	Params       map[string]interface{} `json:"params"`
+}
+
+var traceRecords []TraceRecord
+var traceMutex sync.Mutex
+
+// TraceFunction records the trace of the function with the given name and dynamic parameters
+func TraceFunction(functionName string, params map[string]interface{}) {
+	traceMutex.Lock()
+	traceRecords = append(traceRecords, TraceRecord{
+		FunctionName: functionName,
+		Timestamp:    time.Now().UTC(),
+		Params:       params,
+	})
+	traceMutex.Unlock()
+}
+
+func createTraceDataJSON(outputFile string, traceRecords []TraceRecord) error {
+	jsonData, err := json.MarshalIndent(traceRecords, "", "  ")
+	if err != nil {
+		return fmt.Errorf("error marshalling JSON: %v", err)
+	}
+
+	err = os.WriteFile(outputFile, jsonData, 0644)
```

Comment:
We need that the group and others can only read and not only the owner because this load-tests runs in CI
and this file is stored with openshift-ci artifacts.